lib/ecco/row_converter.rb

```diff
@@ -0,0 +1,61 @@
+module Ecco
+  module RubyConverter
+    FIRST_COLUMN = 0
+    SECOND_COLUMN = 1
+    KEY = 0
+    VALUE = 1
+    ROW = 0
+
+    def self.to_ruby(java_rows, type)
+      if type == "UPDATE_ROWS"
+        java_array_list_to_ruby_array(java_rows)
+      else
+        java_linked_list_to_ruby_array(java_rows)
+      end
+    end
+
+    def self.java_linked_list_to_ruby_array(java_rows)
```

Comment:
I think that things potentially can be simplified:

```
[1] pry(main)> java_list = Java::JavaUtil::LinkedList.new
=> #<Java::JavaUtil::LinkedList:0x4372b9b6>
[2] pry(main)> ruby_list = java_list.to_a
=> []
[3] pry(main)> ruby_list.class
=> Array
[4] pry(main)> java_list.class
=> Java::JavaUtil::LinkedList
```

I haven't tested more than the above though.

Reply:
Interesting, I will look more into this as well.

Reply:
Changed here: d3821566dc513a7d40e88c805a50f45202091f19